crates/wallet/src/wallet/mod.rs

```diff
@@ -1417,8 +1417,33 @@ impl Wallet {
 
         fee_amount += fee_rate * tx.weight();
 
-        let (required_utxos, optional_utxos) =
-            self.preselect_utxos(&params, Some(current_height.to_consensus_u32()));
+        let (required_utxos, optional_utxos) = {
+            // NOTE: manual selection overrides unspendable
+            let mut required: Vec<WeightedUtxo> = params
+                .utxos
+                .clone()
+                .into_iter()
+                // transform required UTxOs into a WeightedUtxo
+                .map(|utxo| WeightedUtxo {
+                    satisfaction_weight: self
+                        .public_descriptor(utxo.keychain)
+                        .max_weight_to_satisfy()
+                        .unwrap(),
+                    utxo: Utxo::Local(utxo),
+                })
+                // include foreign UTxOs as required
+                .chain(params.foreign_utxos.clone().into_values())
+                .collect();
+            let optional = self.filter_utxos(&params, current_height.to_consensus_u32());
```

Comment:
So `optional` is now a subset of `required`? If so, `required` is really `all` utxos?